packages/access-api/src/api.ts

```diff
@@ -0,0 +1,66 @@
+/* eslint-disable @typescript-eslint/method-signature-style */
+/* eslint-disable unicorn/prefer-export-from */
+import type {
+  DID,
+  DIDKey as SpaceDID,
+  Failure,
+  Unit,
+  Result,
+  ServiceMethod,
+  InferInvokedCapability,
+} from '@ucanto/interface'
+import * as UploadAPI from './service/upload-api-proxy.js'
+import * as AccessAPI from '@web3-storage/access/types'
+import type { RouteContext } from './bindings.js'
+import * as Capabilities from '@web3-storage/capabilities'
+
+export * from '@ucanto/interface'
+export * from '@web3-storage/capabilities/types'
+export type { SpaceDID, Unit, RouteContext }
+export type AccountDID = DID<'mailto'>
+export type ServiceDID = DID<'web'>
+
+export type { ProvisionsStorage } from './types/provisions.js'
+export type { DelegationsStorage } from './types/delegations.js'
+
+export type AllocationError = InsufficientStorage
+
+export interface InsufficientStorage extends Failure {
+  name: 'InsufficientStorage'
+}
+
+export interface SpaceProviderRegistry {
+  hasStorageProvider(space: SpaceDID): Promise<Result<boolean, never>>
+}
+
+export { AccessAPI, UploadAPI }
+export interface Service extends AccessAPI.Service, UploadAPI.Service {
+  consumer: {
+    has: ServiceMethod<
+      InferInvokedCapability<typeof Capabilities.Consumer.has>,
+      boolean,
+      Failure
+    >
+  }
+  customer: {
+    get: ServiceMethod<
+      InferInvokedCapability<typeof Capabilities.Customer.get>,
+      CustomerGetOk,
+      CustomerGetError
+    >
+  }
+}
```

Comment:
> nit: should we add this in follow up PR instead of with the upgrade PR?

The problem is that tests were reaching into internals to check state after e.g. authorization which introduced need to thread through internals into tests and a lot of complicated layers.

I have changed tests to make them linear and avoid having to pass bunch of internals. Instead I've exposed things that they need to check as the above capabilities that only server / provider can invoke. So we either keep tests messy and deal with bunch of typing issues or we remove all that and add some capabilities so they can be invoked in tests to check for state.